packages/sdk/src/client.ts

```diff
@@ -736,6 +721,148 @@ export function SoundClient({
+    editionAddress,
+  }: {
+    editionAddress: string
+  }): Promise<string[] | null> {
+    const soundSubgraph = client.soundSubgraph
+
+    if (!soundSubgraph) return null
+
+    const { data, errors } = await soundSubgraph.minterInfo({ editionAddress })
+    const minterInfo = data?.songs
+
+    // if we cannot get any data from the subgraph we return null
+    if (!minterInfo) return null
+    // if subgraph throws an error we return null
+    if (errors) return null
+
+    const editionMinters = minterInfo
+      // just to to be safe we only keep the address we have
+      .filter((edition) => edition.address.toLowerCase() === editionAddress.toLowerCase())
+      // Filter out any falsy values
+      .filter((edition) => Boolean(edition.minters))
+      // Minters are an array so we have to flatten them out
+      .flatMap((edition) => edition.minters)
+
+    if (!editionMinters) {
+      throw new UnexpectedApiResponse({
+        message: 'Minter Information could not be found',
+        graphqlErrors: errors,
+      })
+    }
+
+    const minterAddresses = editionMinters.map((minter) => minter?.address).filter(Boolean) as string[]
+
+    // Filter out any duplicates
+    return Array.from(new Set(minterAddresses))
+  }
+
+  async function _editionMinterMintIdsFromSubgraph({
+    editionAddress,
+    minterAddress,
+  }: {
+    editionAddress: string
+    minterAddress: string
+  }) {
+    const soundSubgraph = client.soundSubgraph
+    if (!soundSubgraph) return null
+
+    const { data, errors } = await soundSubgraph.minterInfo({ editionAddress })
+    const minterInfo = data?.songs
+
+    // if we cannot get any data from the subgraph we return null
+    if (!minterInfo) return null
+    // subgraph throws an error we don't want to throw an error
+    if (errors) return null
+
+    const editionMinters = minterInfo
+      // just to to be safe we only keep the address we have
+      .filter((edition) => edition.address.toLowerCase() === editionAddress.toLowerCase())
+      // Filter out any falsy values
+      .filter((edition) => Boolean(edition.minters))
+      // Minters are an array so we have to flatten them out
+      .flatMap((edition) => edition.minters)
+
+    if (!editionMinters) {
+      throw new UnexpectedApiResponse({
+        message: 'Minter Information could not be found',
+        graphqlErrors: errors,
+      })
+    }
+
+    const mintIds = editionMinters
+      // we only want mint IDs for a given minter address
+      // NOTE: AND/OR filters in graph are not supported yet once they do that then we can write another query
+      .filter((minter) => minter?.address.toLowerCase() === minterAddress.toLowerCase())
+      .map((minter) => minter?.mintId)
+      .filter(Boolean) as string[]
+
+    // Filter out any duplicates
+    return Array.from(new Set(mintIds.map((mintId) => BigNumber.from(mintId).toNumber())))
+  }
+
+  async function _editionMinterMintIdsFromChain({
```

Comment:
NIT: could you group the chain functions and subgraph functions together? Or if you want to group by what they're fetching, go chain, subgraph, chain, subgraph, etc